venom_test.go

```diff
@@ -0,0 +1,87 @@
+package occam_test
+
+import (
+	"fmt"
+	"testing"
+
+	. "github.com/onsi/gomega"
+	"github.com/paketo-buildpacks/occam"
+	"github.com/paketo-buildpacks/occam/fakes"
+	"github.com/paketo-buildpacks/packit/v2/pexec"
+	"github.com/sclevine/spec"
+)
+
+func testVenom(t *testing.T, context spec.G, it spec.S) {
+	var (
+		Expect = NewWithT(t).Expect
+
+		executable *fakes.Executable
+		venom      occam.Venom
+	)
+
+	it.Before(func() {
+		executable = &fakes.Executable{}
+		executable.ExecuteCall.Stub = func(execution pexec.Execution) error {
+			fmt.Fprintln(execution.Stdout, "some stdout output")
+			fmt.Fprintln(execution.Stderr, "some stderr output")
+			return nil
+		}
+
+		venom = occam.NewVenom().WithExecutable(executable)
+	})
+
+	context("Venom", func() {
+		it("should work", func() {
+			logs, err := venom.Execute("test.yaml")
+			Expect(err).NotTo(HaveOccurred())
+			Expect(logs).To(Equal("some stdout output\nsome stderr output\n"))
+
+			Expect(executable.ExecuteCall.Receives.Execution.Args).To(Equal([]string{
+				"run",
+				"test.yaml",
+			}))
+		})
+
+		context("WithVerbose", func() {
+			it("should add -vv flag", func() {
+				_, err := venom.WithVerbose().Execute("test.yaml")
+				Expect(err).NotTo(HaveOccurred())
+
+				Expect(executable.ExecuteCall.Receives.Execution.Args).To(Equal([]string{
+					"run",
+					"-vv",
+					"test.yaml",
+				}))
+			})
+		})
+		context("WithPort", func() {
+			it("should add port variable", func() {
+				_, err := venom.WithPort("8080").Execute("test.yaml")
+				Expect(err).NotTo(HaveOccurred())
+
+				Expect(executable.ExecuteCall.Receives.Execution.Args).To(Equal([]string{
+					"run",
+					"--var",
+					"port=8080",
+					"test.yaml",
+				}))
+			})
+		})
+
+		context("WithVars", func() {
+			it("should add 2 variables", func() {
```

Comment:
I'd probably suggest renaming this test to: `"it should preserve order of provided vars"` and I would add another one or two variables.

That way you're (statistically) demonstrating the ordering is preserved. Adding another one or two keys increases the chance that the test would fail if ordering was random.

Reply:
Done. Although I think the order would not even make any difference. It just adds this variables to be usable in the venom file.